org.eclipse.jdt.core.javac/src/org/eclipse/jdt/core/dom/JavacBindingResolver.java

```diff
@@ -1422,6 +1422,17 @@ public Types getTypes() {
 		return Types.instance(this.context);
 	}
 
+	@Override
+	ITypeBinding getTypeBinding(org.eclipse.jdt.internal.compiler.lookup.TypeBinding referenceBinding) {
```

Comment:
How come that twe have some ECJ types being used here? The JavacBindingResolver should usually not be called with ECJ types.

Reply:
Apparently the .compiler.lookup.* classes are an older version of the bindings. 

Logic flows to this method with the following stack:
```
JavacBindingResolver.getTypeBinding(TypeBinding) line: 1427	
TypeBinding.getErasure() line: 466	
DOMCompletionEngine.findInSupers(ITypeBinding, ITypeBinding) line: 1211	
DOMCompletionEngine.toConstructorProposals(ITypeBinding, ASTNode) line: 1489	
DOMCompletionEngine.completeConstructor(ITypeBinding, ASTNode, IJavaProject) line: 1016	
DOMCompletionEngine.run() line: 611	
CompilationUnit.codeComplete(int, CompletionRequestor, WorkingCopyOwner, IProgressMonitor) line: 461	
CompilationUnit.codeComplete(int, CompletionRequestor, WorkingCopyOwner) line: 455	
```

Reply:
So it seems like we have a mix of TypeBindings from ECJ and Javac here? I suspect that the current DOM is from Javac and the ECJ ones come from the search?

Reply:
![image](https://github.com/user-attachments/assets/ff53460e-94c8-44c9-8f81-662b8eb2bb05)

Seems there's an old one for java.lang.Object.  Not sure where else it might come from.

Reply:
OK, the thing is that we really want to avoid dealing with ECJ bindings in general. I'll try to have a look at it later, please do not merge in the meantime.

Reply:
This is a signature in the superclass, "BindingResolver". We really can't remove the dependency on the class. And if you check, all we're actually doing is finding a binding with the same signature in our cache. 

But I can't change the superclass signature. 

I genuinely believe this is a minimal dependency and we should maintain it. We're certainly not reaching into actual functionality here.